src/fan.ml

```diff
@@ -44,9 +44,11 @@ let nb_fans t = Array.length t.fans
 let fan t h = (h land t.mask) lsr t.shift
 
 let search t h =
-  let fan = fan t h in
-  let low = if fan = 0 then 0L else t.fans.(fan - 1) in
-  (low, t.fans.(fan))
+  if nb_fans t = 0 then (0L, 0L)
```

Comment:
I think this upper bound is going to be interpreted inclusively by `interpolation_search`, so this is not correct.

Reply:
Good catch, thank you! I modified it to `(0L, 1L)`. (It's a bit annoying that no test failed because of this.)

Reply:
Hmm, shouldn't it be `(0L, -1L)` if you want to make the range empty?

Reply:
> Good catch, thank you! I modified it to (0L, 1L). (It's a bit annoying that no test failed because of this.)

Sorry, made a bad manipulation :)

Reply:
> Hmm, shouldn't it be `(0L, -1L)` if you want to make the range empty?

Yes, indeed, but the failing test passed, I'll look more into it...

Reply:
I couldn't reproduce the failing test for this (I think it is fixed by https://github.com/mirage/index/pull/168/files#diff-732d381615307f94d510e2dbdee74f4aR483) so I removed it.